File: frontend/src/app/classes/session-started-notification.ts
import {Notification} from './notification';
import {User} from './user';
import {FTSession} from './FTSession';
import {Course} from './course';

export class SessionStartedNotification extends Notification {
  session: FTSession;

  constructor(id: number, message: string, user: User, session: FTSession, course: Course) {
    super(id, message, user, course, `A new session has started!`);
    this.session = session;
  }
}

File: backend/src/main/java/com/fullteaching/backend/model/SessionStartedNotification.java
package com.fullteaching.backend.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Entity;
import javax.persistence.ManyToOne;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class SessionStartedNotification extends Notification {

    @ManyToOne
    private Session session;

    public SessionStartedNotification(String message, User user, Session session) {
        super(message, user, session.getCourse(), NotificationType.SESSION_STARTED);
        this.session = session;
    }
}
